File: fsharp/collatz-conjecture/CollatzConjecture.fs
module CollatzConjecture

let steps (number: int): int option = 
    let mutable x = number
    let mutable count = 0
    while x <> 1 do
        printfn "%d" x
        if x % 2 = 0 then
            x <- x / 2
        elif x % 2 = 1 then
            x <- 3 * x + 1
        count <- count + 1    
    Some(count)
    

File: solutions/scala/collatz-conjecture/1/src/main/scala/CollatzConjecture.scala
object CollatzConjecture {
  def steps(initial: Int, count: Int = 0): Option[Int] = initial match {
    case x if x < 1 => None
    case x if x == 1 => Some(count)
    case x if x % 2 == 0 => steps(x >> 1, count + 1)
    case x => steps((3 * x) + 1, count + 1)
  }
}